modules/solid_mechanics/src/actions/LineElementAction.C

```diff
@@ -486,7 +487,8 @@ LineElementAction::actAddVariables()
       for (const auto & rot : _rotations)
       {
         // Create rotation variables
-        _problem->addVariable("MooseVariable", rot, params);
+        if (!_problem->hasVariable(rot))
```

Comment:
we might have to sprinkle a few of these in actions that have the intent of adding these variables fo rthe user but users might want a little more restriction. this is likely not the only one.